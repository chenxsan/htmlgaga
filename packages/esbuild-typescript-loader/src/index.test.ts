import path from 'path'
import webpack, { Stats } from 'webpack'
import { createFsFromVolume, Volume } from 'memfs'
function webpackCompiler(fixture: string, options = {}): Promise<Stats> {
  const compiler = webpack({
    context: __dirname,
    entry: `./${fixture}`,
    output: {
      path: path.resolve(__dirname),
      filename: 'bundle.js',
    },
    module: {
      rules: [
        {
          test: /\.ts$/,
          use: [
            {
              loader: path.resolve(__dirname, 'index.ts'),
              options,
            },
          ],
        },
        {
          test: /\.tsx$/,
          use: [
            {
              loader: path.resolve(__dirname, 'index.ts'),
              options,
            },
          ],
        },
      ],
    },
  })
  // eslint-disable-next-line
  // @ts-ignore
  compiler.outputFileSystem = createFsFromVolume(new Volume())
  return new Promise((resolve, reject) => {
    compiler.run((err, stats) => {
      if (err) reject(err)
      if (stats?.hasErrors()) reject(stats.toJson().errors)
      if (typeof stats !== 'undefined') resolve(stats)
    })
  })
}
describe('esbuild TypeScript Loader', () => {
  it('should transform ts', async () => {
    const stats = await webpackCompiler('example.ts')
    const modules = stats.toJson({ source: true }).modules ?? []
    const output = modules[0].source
    expect(output).toContain('const a = 42;')
    expect(output).toMatchSnapshot()
  })
  it('should transform tsx', async () => {
    const stats = await webpackCompiler('example.tsx', {
      loader: 'tsx',
    })
    const modules = stats.toJson({ source: true }).modules ?? []
    const output = modules[0].source
    expect(output).toMatchSnapshot()
  })
})
