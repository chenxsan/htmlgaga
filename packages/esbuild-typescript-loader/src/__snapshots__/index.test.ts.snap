// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`esbuild TypeScript Loader should transform ts 1`] = `
"const a = 42;
const user = {
  name: \\"Hayes\\",
  id: 0
};
export {a, user};
"
`;

exports[`esbuild TypeScript Loader should transform tsx 1`] = `
"export default function Example(props) {
  return /* @__PURE__ */ React.createElement(\\"div\\", null, props.title);
}
"
`;
